packages/lesswrong/server/resolvers/aiArtResolvers/coverImageMutations.ts

```diff
@@ -0,0 +1,66 @@
+import { generateCoverImagesForPost } from '@/server/scripts/generativeModels/coverImages-2023Review';
+import { userIsAdmin } from '@/lib/vulcan-users/permissions';
+import SplashArtCoordinates from '@/server/collections/splashArtCoordinates/collection';
+import ReviewWinnerArts from '@/server/collections/reviewWinnerArts/collection';
+import { gql } from 'apollo-server';
+
+export const generateCoverImagesForPostGraphQLTypeDefs = gql`
+  extend type Mutation {
+    generateCoverImagesForPost(postId: String!, prompt: String): [ReviewWinnerArt]
+  }
+`
+
+export const generateCoverImagesForPostGraphQLMutations = {
+  generateCoverImagesForPost: async (root: void, { postId, prompt }: { postId: string, prompt?: string }, context: ResolverContext) => {
+    const { currentUser } = context;
+    if (!currentUser || !userIsAdmin(currentUser)) {
+      throw new Error('You must be an admin to generate cover images');
+    }
+    try {
+      const results = await generateCoverImagesForPost(postId, prompt);
+      const reviewWinnerArts = results.map(result => result.reviewWinnerArt);
+      return reviewWinnerArts;
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error('Error generating cover images:', error);
+      throw new Error(`Error generating cover images: ${error.message}`);
+    }
+  },
+}
+
+export const flipSplashArtImageGraphQLTypeDefs = gql`
+  extend type Mutation {
+    flipSplashArtImage(reviewWinnerArtId: String!): Boolean
+  }
+`
+
+export const flipSplashArtImageGraphQLMutations = {
+  flipSplashArtImage: async (root: void, { reviewWinnerArtId }: { reviewWinnerArtId: string }, context: ResolverContext) => {
+    const currentSplashCoordinates = await SplashArtCoordinates.findOne({ reviewWinnerArtId }, { sort: { createdAt: -1 } });
+    const reviewWinnerArt = await ReviewWinnerArts.findOne({ _id: reviewWinnerArtId });
```

Comment:
not remotely a blocker (particularly for admin-only functionality) but these could be in a Promise.all